PT-3078: implemented registration form revamp

captaincrazybro opened the pull request:
List of changes:
- Made a separate web view and command for the internet settings
- Made a separate web view and command for the paratext registry profile (different than the initial paratext registration screen)
- Changed the tab icons for the registry and internet settings
- Added intermediate stage for viewing the registration details
- Added additional text for paratext registration page and profile page
- Changed sizing of the tabs

Initial paratext registration popup:
<img width="642" height="556" alt="image" src="https://github.com/user-attachments/assets/7db2e287-5dfe-4bff-b3b3-9be4756a68e5" />

New tab icons:
<img width="236" height="104" alt="image" src="https://github.com/user-attachments/assets/b3b69326-c7bd-4032-8e79-8557a22db5fa" />

New profile page:
<img width="656" height="581" alt="image" src="https://github.com/user-attachments/assets/7aedb74c-c8f5-4d64-9842-996163bb46ea" />

Editing profile:
<img width="656" height="581" alt="image" src="https://github.com/user-attachments/assets/35a053d2-25cf-4c38-9d5a-909992bcfa0e" />

Registration code too short:
<img width="621" height="567" alt="image" src="https://github.com/user-attachments/assets/167d3073-215c-40fd-8011-7d0086497815" />

Registration code has invalid character(s):
<img width="621" height="567" alt="image" src="https://github.com/user-attachments/assets/cd5a5d81-e49f-42d9-9982-87e05609a8ff" />

Invalid registration code format:
<img width="621" height="567" alt="image" src="https://github.com/user-attachments/assets/c614b300-83bf-4a2f-8e69-807014f08b77" />


Registration code and name doesn't match:
<img width="656" height="581" alt="image" src="https://github.com/user-attachments/assets/e514c99b-5d18-41d0-8013-980ec226efa1" />

Valid registration code (before saving):
<img width="520" height="471" alt="image" src="https://github.com/user-attachments/assets/449c80c4-1ee3-47b5-815b-3055f24fba32" />

Registration code and name hasn't been modified:
<img width="500" height="465" alt="image" src="https://github.com/user-attachments/assets/d7962b81-0253-4ebf-a856-f8d4040d7074" />

When registration code and name have first been saved (in the midst of restarting):
<img width="500" height="465" alt="image" src="https://github.com/user-attachments/assets/620320fa-384b-4ee8-b0b5-1e43323a30f6" />

Save and restart successful:
<img width="662" height="597" alt="image" src="https://github.com/user-attachments/assets/74fdbdd8-e65e-4c90-9c43-5b77f6c9743f" />

Internet settings popup:
<img width="558" height="533" alt="image" src="https://github.com/user-attachments/assets/a2673de2-94da-4fa6-82dc-2d2417516b75" />


<!-- Reviewable:start -->
- - -
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/paranext/paranext-core/1837)
<!-- Reviewable:end -->

tjcouch-sil reviewed (changes requested):
Thanks for the nice changes! I only reviewed files on which I'm codeowner and a couple other points of interest; will need another review as well.

> @tjcouch-sil reviewed 3 of 18 files at r1.
> *[Reviewable](https://reviewable.io/reviews/paranext/paranext-core/1837#-:-OZiqTLy8zragyw8CM-i:b-a6wzfj)* status: 3 of 18 files reviewed, 4 unresolved discussions

___
*[`extensions/src/paratext-registration/contributions/localizedStrings.json` line 82 at r1](https://reviewable.io/reviews/paranext/paranext-core/1837#-OZilaXmAJzWqf2dP8cK:-OZilaXmAJzWqf2dP8cL:b-dvv24x) ([raw file](https://github.com/paranext/paranext-core/blob/373afef728852e12e24ac8f6d15b813099442073/extensions/src/paratext-registration/contributions/localizedStrings.json#L82)):*
> ```JSON
>       "%paratextRegistration_section_proxySettings%": "Proxy Settings",
>       "%paratextRegistration_warning_invalid_email%": "Please enter a valid email address.",
>       "%paratextRegistration_warning_invalid_registration%": "This registration code is not correct for this user.",
> ```

According to our [localized strings guidelines](https://github.com/paranext/paranext/wiki/Code-Style-Guide#localized-strings-guidelines), we need to preserve existing strings that were already published. You can mark this old string key `%paratextRegistration_warning_invalid_registration%` deprecated if you'd like. There are links in the guidelines for how to deprecate a string.
___
*[`extensions/src/paratext-registration/contributions/localizedStrings.json` line 83 at r1](https://reviewable.io/reviews/paranext/paranext-core/1837#-OZilw0uBAVsZlSh4gey:-OZilw0vEYvN8BTOzRjK:bv92huz) ([raw file](https://github.com/paranext/paranext-core/blob/373afef728852e12e24ac8f6d15b813099442073/extensions/src/paratext-registration/contributions/localizedStrings.json#L83)):*
> ```JSON
>       "%paratextRegistration_warning_invalid_email%": "Please enter a valid email address.",
>       "%paratextRegistration_warning_invalid_registration%": "This registration code is not correct for this user.",
>       "%paratextRegistration_webView_title%": "Paratext Registration Information",
> ```

According to our [localized strings guidelines](https://github.com/paranext/paranext/wiki/Code-Style-Guide#localized-strings-guidelines), we need to preserve existing strings that were already published. You can mark this string key `%paratextRegistration_webView_title%` deprecated and make a new one `%paratextRegistration_webView_title_2%` if you'd like. There are links in the guidelines for how to deprecate a string.

An illustration of why this is important is here: see that this string is localized into Spanish already. If you change the English now, the Spanish will be out of date. We don't have control over all localizations that have ever been done because they are extensible, so we don't know if someone else has also localized this string into another language.
___
*[`extensions/src/paratext-registration/src/types/paratext-registration.d.ts` line 72 at r1](https://reviewable.io/reviews/paranext/paranext-core/1837#-OZin2c828o5D8zyQpzL:-OZin2c828o5D8zyQpzM:bhcwuyl) ([raw file](https://github.com/paranext/paranext-core/blob/4733ce2db1fe2bf65f703afd480f742f4acf09b3/extensions/src/paratext-registration/src/types/paratext-registration.d.ts#L72)):*
> ```XML
>      * @returns Id of the internet settings web view
>      */
>     'paratextRegistration.showInternetSettings': () => Promise<string | undefined>;
> ```

I don't see any screenshots of this WebView FYI. Might make it easier for UX to review to include some screenshots of it as well :)
___
*[`extensions/src/paratext-registration/src/types/paratext-registration.d.ts` line 79 at r1](https://reviewable.io/reviews/paranext/paranext-core/1837#-OZikYG42sthE6U1kcXq:-OZikYG565pm4EU0nCb2:b-tzedk9) ([raw file](https://github.com/paranext/paranext-core/blob/4733ce2db1fe2bf65f703afd480f742f4acf09b3/extensions/src/paratext-registration/src/types/paratext-registration.d.ts#L79)):*
> ```XML
>      * @returns Id of the profile web view
>      */
>     'paratextRegistration.showProfile': () => Promise<string | undefined>;
> ```

What's the difference between this and `paratextRegistration.showParatextRegistration`? Could you include some screenshots and explanation of what different use these two views have and in what different ways they would be accessed?

If they're extremely similar, could the profile WebView replace the registration WebView and we just keep the old registration command and not have to add a new `showProfile` command?
___
*[`extensions/src/platform-get-resources/src/home.web-view.tsx` line 158 at r1](https://reviewable.io/reviews/paranext/paranext-core/1837#-OZiofrl5MBaJXrkEl6N:-OZiofrl5MBaJXrkEl6O:b-oe4ua) ([raw file](https://github.com/paranext/paranext-core/blob/4733ce2db1fe2bf65f703afd480f742f4acf09b3/extensions/src/platform-get-resources/src/home.web-view.tsx#L158)):*
> ```XML
>             message: '%data_loading_error_paratextData_internet_disabled%',
>             clickCommandLabel: '%general_open%',
>             clickCommand: 'paratextRegistration.showInternetSettings',
> ```

Ah this seems like a nice change. But unfortunately this is a copypasta around our repos, so there are a few other places this needs to be changed... :grimacing: 

I did [a search in our org for `showParatextRegistration`](https://github.com/search?q=org%3Aparanext%20showParatextRegistration&type=code) and found some other spots of interest that may need changing:

[paratext-bible-internal-extensions/src/paratext-bible-send-receive/src/send-receive.web-view.tsx at 62b855c4a321d2355b2ca0145ebe12c23d41965c · paranext/paratext-bible-internal-extensions](https://github.com/paranext/paratext-bible-internal-extensions/blob/62b855c4a321d2355b2ca0145ebe12c23d41965c/src/paratext-bible-send-receive/src/send-receive.web-view.tsx#L129)

[paratext-bible-internal-extensions/src/paratext-bible-send-receive/src/components/send-receive-view.component.tsx at 62b855c4a321d2355b2ca0145ebe12c23d41965c · paranext/paratext-bible-internal-extensions](https://github.com/paranext/paratext-bible-internal-extensions/blob/62b855c4a321d2355b2ca0145ebe12c23d41965c/src/paratext-bible-send-receive/src/components/send-receive-view.component.tsx#L136)

[paratext-bible-internal-extensions/src/paratext-bible-send-receive/src/main.ts at 62b855c4a321d2355b2ca0145ebe12c23d41965c · paranext/paratext-bible-internal-extensions](https://github.com/paranext/paratext-bible-internal-extensions/blob/62b855c4a321d2355b2ca0145ebe12c23d41965c/src/paratext-bible-send-receive/src/main.ts#L220)

[paranext-core/extensions/src/paratext-registration/contributions/menus.json at b3bfce647de39c4bbf47a530c40c7dd38b67757d · paranext/paranext-core](https://github.com/paranext/paranext-core/blob/b3bfce647de39c4bbf47a530c40c7dd38b67757d/extensions/src/paratext-registration/contributions/menus.json#L11) - did you mean to leave this one as `showParatextRegistration`? I imagine you did, but I'm confused as to why since (as I mentioned in another comment) I don't know the difference between the two, and it seems you pretty much exclusively replaced `showParatextRegistration` with `showProfile`.
___
*[`src/shared/services/data-provider.service.ts` line 119 at r1](https://reviewable.io/reviews/paranext/paranext-core/1837#-OZin_6M5BI-IkRRYgX6:-OZin_6M5BI-IkRRYgX7:b9pbyvh) ([raw file](https://github.com/paranext/paranext-core/blob/4733ce2db1fe2bf65f703afd480f742f4acf09b3/src/shared/services/data-provider.service.ts#L119)):*
> ```XML
>   if (isErrorMessageAboutParatextBlockingInternetAccess(exception)) {
>     retVal.message = '%data_loading_error_paratextData_internet_disabled%';
>     retVal.clickCommand = 'paratextRegistration.showInternetSettings' as keyof CommandHandlers;
> ```

You'll probably need these lines above this line:

```ts
// TS doesn't realize this is a valid command handler key since it is defined in an extension
// eslint-disable-next-line no-type-assertion/no-type-assertion
```

We don't allow type assertion unless it's necessary, and we require commenting with an explanation of why it's necessary. That's why we have the eslint rule.


<!-- Sent from Reviewable.io -->

captaincrazybro commented:
*[`extensions/src/paratext-registration/src/types/paratext-registration.d.ts` line 79 at r1](https://reviewable.io/reviews/paranext/paranext-core/1837#-OZikYG42sthE6U1kcXq:-OZjl9xU2U48stgGvkSa:b-bzj982) ([raw file](https://github.com/paranext/paranext-core/blob/4733ce2db1fe2bf65f703afd480f742f4acf09b3/extensions/src/paratext-registration/src/types/paratext-registration.d.ts#L79)):*
<details><summary><i>Previously, tjcouch-sil (TJ Couch) wrote…</i></summary><blockquote>

What's the difference between this and `paratextRegistration.showParatextRegistration`? Could you include some screenshots and explanation of what different use these two views have and in what different ways they would be accessed?

If they're extremely similar, could the profile WebView replace the registration WebView and we just keep the old registration command and not have to add a new `showProfile` command?
</blockquote></details>

Basically, the profile is meant to be shown if the registration has already been set (a way of editing the existing registration) and the regular `showParatextRegistration` command shows the initial paratext registration screen. More information can be found here: https://www.figma.com/design/IlfaNPYLjHFFMnR8V7oD3j/Registration-PT-3078?node-id=55-1908&t=Izl9ChkvkUwmyqCz-0


<!-- Sent from Reviewable.io -->

captaincrazybro reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/paranext/paranext-core/1837)* status: 1 of 18 files reviewed, 4 unresolved discussions (waiting on @tjcouch-sil)

___
*[`extensions/src/paratext-registration/contributions/localizedStrings.json` line 82 at r1](https://reviewable.io/reviews/paranext/paranext-core/1837#-OZilaXmAJzWqf2dP8cK:-OZjl7ofEAt7MqvmTzFP:b-896fix) ([raw file](https://github.com/paranext/paranext-core/blob/373afef728852e12e24ac8f6d15b813099442073/extensions/src/paratext-registration/contributions/localizedStrings.json#L82)):*
<details><summary><i>Previously, tjcouch-sil (TJ Couch) wrote…</i></summary><blockquote>

According to our [localized strings guidelines](https://github.com/paranext/paranext/wiki/Code-Style-Guide#localized-strings-guidelines), we need to preserve existing strings that were already published. You can mark this old string key `%paratextRegistration_warning_invalid_registration%` deprecated if you'd like. There are links in the guidelines for how to deprecate a string.
</blockquote></details>

Done.
___
*[`extensions/src/paratext-registration/contributions/localizedStrings.json` line 83 at r1](https://reviewable.io/reviews/paranext/paranext-core/1837#-OZilw0uBAVsZlSh4gey:-OZjl8Gq8jPIzN_4tMqj:b-896fix) ([raw file](https://github.com/paranext/paranext-core/blob/373afef728852e12e24ac8f6d15b813099442073/extensions/src/paratext-registration/contributions/localizedStrings.json#L83)):*
<details><summary><i>Previously, tjcouch-sil (TJ Couch) wrote…</i></summary><blockquote>

According to our [localized strings guidelines](https://github.com/paranext/paranext/wiki/Code-Style-Guide#localized-strings-guidelines), we need to preserve existing strings that were already published. You can mark this string key `%paratextRegistration_webView_title%` deprecated and make a new one `%paratextRegistration_webView_title_2%` if you'd like. There are links in the guidelines for how to deprecate a string.

An illustration of why this is important is here: see that this string is localized into Spanish already. If you change the English now, the Spanish will be out of date. We don't have control over all localizations that have ever been done because they are extensible, so we don't know if someone else has also localized this string into another language.
</blockquote></details>

Done.


<!-- Sent from Reviewable.io -->

tjcouch-sil reviewed (changes requested):
> @tjcouch-sil reviewed 3 of 7 files at r2.
> *[Reviewable](https://reviewable.io/reviews/paranext/paranext-core/1837)* status: 4 of 18 files reviewed, 2 unresolved discussions (waiting on @captaincrazybro)

___
*[`extensions/src/paratext-registration/contributions/localizedStrings.json` line 82 at r1](https://reviewable.io/reviews/paranext/paranext-core/1837#-OZilaXmAJzWqf2dP8cK:-OZkJAsR68xhJ1UAqylk:b34gtvp) ([raw file](https://github.com/paranext/paranext-core/blob/373afef728852e12e24ac8f6d15b813099442073/extensions/src/paratext-registration/contributions/localizedStrings.json#L82)):*
<details><summary><i>Previously, captaincrazybro (Cqptain) wrote…</i></summary><blockquote>

Done.
</blockquote></details>

Nice! Thanks :)
___
*[`extensions/src/paratext-registration/contributions/localizedStrings.json` line 83 at r1](https://reviewable.io/reviews/paranext/paranext-core/1837#-OZilw0uBAVsZlSh4gey:-OZkJI5P3e1itUc4foyR:b-4i6eqh) ([raw file](https://github.com/paranext/paranext-core/blob/373afef728852e12e24ac8f6d15b813099442073/extensions/src/paratext-registration/contributions/localizedStrings.json#L83)):*
<details><summary><i>Previously, captaincrazybro (Cqptain) wrote…</i></summary><blockquote>

Done.
</blockquote></details>

It doesn't seem from my current view that anything changed about this string. Am I missing something?
___
*[`extensions/src/paratext-registration/src/types/paratext-registration.d.ts` line 79 at r1](https://reviewable.io/reviews/paranext/paranext-core/1837#-OZikYG42sthE6U1kcXq:-OZkJfE93mAwIhpBOjDt:b7xzt1m) ([raw file](https://github.com/paranext/paranext-core/blob/4733ce2db1fe2bf65f703afd480f742f4acf09b3/extensions/src/paratext-registration/src/types/paratext-registration.d.ts#L79)):*
<details><summary><i>Previously, captaincrazybro (Cqptain) wrote…</i></summary><blockquote>

I did realize though that my logic has some flaws. So I will be making just one command but two web views that are shown conditionally.
</blockquote></details>

Instead of having two separate WebViews, would it be reasonably possible to make it a little simpler by having the two be the same WebView and just change the title and move a couple things around based on whether the registration has already been set? Seems like a lot of extra code running and memory used for such a little difference.

I'm not deep in the discussions around this work, though, so this thread is not blocking.
___
*[`extensions/src/platform-get-resources/src/home.web-view.tsx` line 158 at r1](https://reviewable.io/reviews/paranext/paranext-core/1837#-OZiofrl5MBaJXrkEl6N:-OZkKiUF1ttU3Hb7wXgs:b6eg23o) ([raw file](https://github.com/paranext/paranext-core/blob/4733ce2db1fe2bf65f703afd480f742f4acf09b3/extensions/src/platform-get-resources/src/home.web-view.tsx#L158)):*
<details><summary><i>Previously, captaincrazybro (Cqptain) wrote…</i></summary><blockquote>

https://github.com/paranext/paratext-bible-internal-extensions/pull/154
</blockquote></details>

Nice. Thanks!
___
*[`extensions/src/paratext-registration/contributions/localizedStrings.json` line 50 at r2](https://reviewable.io/reviews/paranext/paranext-core/1837#-OZkJ-NK9jw2LWPCgrb5:-OZkJ-NK9jw2LWPCgrb6:b-spzqc0) ([raw file](https://github.com/paranext/paranext-core/blob/b82017687b05b21045d1aee562a270708fe5ae6a/extensions/src/paratext-registration/contributions/localizedStrings.json#L50)):*
> ```JSON
>     "%paratextRegistration_warning_invalid_registration%": {
>       "deprecationInfo": {
>         "message": "The invalid registration warning became an alert with both a title and subtitle instead of just a one-line warning",
> ```

Would recommend listing the key that is closest replacement so people can consider using it :)


<!-- Sent from Reviewable.io -->

captaincrazybro commented:
*[`extensions/src/paratext-registration/contributions/localizedStrings.json` line 83 at r1](https://reviewable.io/reviews/paranext/paranext-core/1837#-OZilw0uBAVsZlSh4gey:-OZkOFJM2OOokqvC1Rn5:bx63z4s) ([raw file](https://github.com/paranext/paranext-core/blob/373afef728852e12e24ac8f6d15b813099442073/extensions/src/paratext-registration/contributions/localizedStrings.json#L83)):*
<details><summary><i>Previously, tjcouch-sil (TJ Couch) wrote…</i></summary><blockquote>

It doesn't seem from my current view that anything changed about this string. Am I missing something?
</blockquote></details>

I don't think I removed this key. I just moved it around. It's still in the `localizedStrings.json` file and we are still using it.


<!-- Sent from Reviewable.io -->

rolfheij-sil reviewed (changes requested):
Yooo, this looks really nice! So much more clear to the user what's going on. Great work.
I do have a bunch of comments though. Sorry, it's a lot :sweat_smile: 

> @rolfheij-sil reviewed 11 of 18 files at r1, 7 of 7 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/paranext/paranext-core/1837#-:-OZnzEjQ2PAzYyauWGr3:bfpxqox)* status: all files reviewed, 15 unresolved discussions (waiting on @captaincrazybro and @tjcouch-sil)

___
*[`extensions/src/paratext-registration/src/types/paratext-registration.d.ts` line 79 at r1](https://reviewable.io/reviews/paranext/paranext-core/1837#-OZikYG42sthE6U1kcXq:-OZneUcZ-usE47nImXzl:b-ddq514) ([raw file](https://github.com/paranext/paranext-core/blob/4733ce2db1fe2bf65f703afd480f742f4acf09b3/extensions/src/paratext-registration/src/types/paratext-registration.d.ts#L79)):*
<details><summary><i>Previously, tjcouch-sil (TJ Couch) wrote…</i></summary><blockquote>

Instead of having two separate WebViews, would it be reasonably possible to make it a little simpler by having the two be the same WebView and just change the title and move a couple things around based on whether the registration has already been set? Seems like a lot of extra code running and memory used for such a little difference.

I'm not deep in the discussions around this work, though, so this thread is not blocking.
</blockquote></details>

Agreed, make sense to put them in the same webview
___
*[`extensions/src/paratext-registration/src/components/section.component.tsx` line 1 at r2](https://reviewable.io/reviews/paranext/paranext-core/1837#-OZnl4k4C92nBTbSTbGk:-OZnl4k4C92nBTbSTbGl:b-nrxk7y) ([raw file](https://github.com/paranext/paranext-core/blob/b82017687b05b21045d1aee562a270708fe5ae6a/extensions/src/paratext-registration/src/components/section.component.tsx#L1)):*
> ```XML
> import { cn } from 'platform-bible-react';
> ```

This component does not make sense to me. Essentially it's just a `div` with `tw-m-2`. Why not just add that Tailwind class instead of creating an entire component for it?
___
*[`extensions/src/paratext-registration/contributions/localizedStrings.json` line 57 at r2](https://reviewable.io/reviews/paranext/paranext-core/1837#-OZnc8DlEPFqsEoUYrzU:-OZnc8DlEPFqsEoUYrzV:bffom7g) ([raw file](https://github.com/paranext/paranext-core/blob/b82017687b05b21045d1aee562a270708fe5ae6a/extensions/src/paratext-registration/contributions/localizedStrings.json#L57)):*
> ```JSON
>   "localizedStrings": {
>     "en": {
>       "%internetSettings_webView_title%": "Internet Settings",
> ```

UX wants us to only use capitalization on the first word. So I believe this should be `Internet settings`

I notice that many strings have words with capitals mid-sentence. Maybe best to ask UX to review all of these strings. Or were they provided by UX?
___
*[`extensions/src/paratext-registration/contributions/localizedStrings.json` line 115 at r2](https://reviewable.io/reviews/paranext/paranext-core/1837#-OZncy-DAeFKe-Pgg9MZ:-OZncy-DAeFKe-Pgg9M_:b-248n2x) ([raw file](https://github.com/paranext/paranext-core/blob/b82017687b05b21045d1aee562a270708fe5ae6a/extensions/src/paratext-registration/contributions/localizedStrings.json#L115)):*
> ```JSON
>     },
>     "es": {
>       "%mainMenu_openInternetSettings%": "Configuración de Internet...",
> ```

Are you sure that this is correct Spanish? Or was this added by AI?
___
*[`extensions/src/paratext-registration/src/components/grid.component.tsx` line 1 at r2](https://reviewable.io/reviews/paranext/paranext-core/1837#-OZnlMJfEgokUJ4FnTxZ:-OZnlMJfEgokUJ4FnTx_:b4zdn9p) ([raw file](https://github.com/paranext/paranext-core/blob/b82017687b05b21045d1aee562a270708fe5ae6a/extensions/src/paratext-registration/src/components/grid.component.tsx#L1)):*
> ```XML
> import { cn } from 'platform-bible-react';
> ```

It this a specific component for this particular extension? Does this help us implement a UX design that occurs more frequently? In other words: Should this be in `platform-bible-react`?

The name `Grid` sounds very generic. Too generic :sweat_smile: 
___
*[`extensions/src/paratext-registration/src/utils/helpers.ts` line 1 at r2](https://reviewable.io/reviews/paranext/paranext-core/1837#-OZnkSXqBOF1JDi-PESK:-OZnkSXr0ykepy9v3EN6:br7nur6) ([raw file](https://github.com/paranext/paranext-core/blob/b82017687b05b21045d1aee562a270708fe5ae6a/extensions/src/paratext-registration/src/utils/helpers.ts#L1)):*
> ```XML
> export function scrollToRef(ref: HTMLElement | null) {
> ```

What do we need this for? I see it is used on `<Alert>` but I don't understand why we need it. Can we try removing it?
___
*[`extensions/src/paratext-registration/src/utils/helpers.ts` line 9 at r2](https://reviewable.io/reviews/paranext/paranext-core/1837#-OZnmTcu-am6IIJ71GKI:-OZnmTcu-am6IIJ71GKJ:blb25oa) ([raw file](https://github.com/paranext/paranext-core/blob/b82017687b05b21045d1aee562a270708fe5ae6a/extensions/src/paratext-registration/src/utils/helpers.ts#L9)):*
> ```XML
> 
> /** Representation of whether the user has saved changes */
> export enum SaveState {
> ```

Maybe we don't need a `utils` folder and we can just place this file in `src` and rename it to `utils.ts`? That more in line with what we do in other places in our codebase
___
*[`extensions/src/paratext-registration/src/components/registration-form.component.tsx` line 25 at r2](https://reviewable.io/reviews/paranext/paranext-core/1837#-OZnm4pzEYeLE2G8ImIX:-OZnm4pzEYeLE2G8ImIY:bnnkztq) ([raw file](https://github.com/paranext/paranext-core/blob/b82017687b05b21045d1aee562a270708fe5ae6a/extensions/src/paratext-registration/src/components/registration-form.component.tsx#L25)):*
> ```XML
>   '^(?:[a-zA-Z0-9]{6}-[a-zA-Z0-9]{6}-[a-zA-Z0-9]{6}-[a-zA-Z0-9]{6}-[a-zA-Z0-9]{6}|\\*{6}-\\*{6}-\\*{6}-\\*{6}-\\*{6})$';
> const REGISTRATION_CODE_INSERT_DASH_REGEX_STRING = '[a-zA-Z0-9]{6}$';
> const EMAIL_REGEX_STRING = '^.+@.+\\..+$';
> ```

This is currently not being used
___
*[`extensions/src/paratext-registration/src/profile.web-view.tsx` line 1 at r2](https://reviewable.io/reviews/paranext/paranext-core/1837#-OZnjfje7iQrU7fZQphh:-OZnjfje7iQrU7fZQphi:bxkjxxf) ([raw file](https://github.com/paranext/paranext-core/blob/b82017687b05b21045d1aee562a270708fe5ae6a/extensions/src/paratext-registration/src/profile.web-view.tsx#L1)):*
> ```XML
> import { WebViewProps } from '@papi/core';
> ```

This web view and the `profile` web view are so similar that I would just  merge them into a single web view. That also removed the need to extract out RegistrationForm and PARATEXT_REGISTRY_LINK  (and maybe more things) into separate files
___
*[`extensions/src/paratext-registration/src/internet-settings.web-view.tsx` line 163 at r2](https://reviewable.io/reviews/paranext/paranext-core/1837#-OZnypgm9p4EoL61p9en:-OZnypgm9p4EoL61p9eo:b-5qnatc) ([raw file](https://github.com/paranext/paranext-core/blob/b82017687b05b21045d1aee562a270708fe5ae6a/extensions/src/paratext-registration/src/internet-settings.web-view.tsx#L163)):*
> ```XML
>       }, SAVE_SETTINGS_DELAY_MS);
>       setSaveTimeout(newSaveTimeout);
>     }
> ```

This useEffect needs to clean up the `saveTimout` timeout when the component unmounts while the timer is still running to prevent leaking memory
___
*[`extensions/src/paratext-registration/src/internet-settings.web-view.tsx` line 164 at r2](https://reviewable.io/reviews/paranext/paranext-core/1837#-OZnvti22Iozy6tIx1kx:-OZnvti22Iozy6tIx1ky:b44d3me) ([raw file](https://github.com/paranext/paranext-core/blob/b82017687b05b21045d1aee562a270708fe5ae6a/extensions/src/paratext-registration/src/internet-settings.web-view.tsx#L164)):*
> ```XML
>       setSaveTimeout(newSaveTimeout);
>     }
>   }, [internetSettings]);
> ```

This seems to be missing some dependencies, right?
___
*[`extensions/src/paratext-registration/src/internet-settings.web-view.tsx` line 242 at r2](https://reviewable.io/reviews/paranext/paranext-core/1837#-OZnwBlx02dEy65HJF6i:-OZnwBlx02dEy65HJF6j:b-tq264t) ([raw file](https://github.com/paranext/paranext-core/blob/b82017687b05b21045d1aee562a270708fe5ae6a/extensions/src/paratext-registration/src/internet-settings.web-view.tsx#L242)):*
> ```XML
>                   setInternetSettings({
>                     ...internetSettings,
>                     proxyPort: e.target.value.length > 0 ? parseInt(e.target.value, 10) : 0,
> ```

parseInt can fail and return NaN. That edge case should be handled too
___
*[`extensions/src/paratext-registration/src/internet-settings.web-view.tsx` line 281 at r2](https://reviewable.io/reviews/paranext/paranext-core/1837#-OZnwY0a6RdUTvSKtLWr:-OZnwY0a6RdUTvSKtLWs:b-kocnyt) ([raw file](https://github.com/paranext/paranext-core/blob/b82017687b05b21045d1aee562a270708fe5ae6a/extensions/src/paratext-registration/src/internet-settings.web-view.tsx#L281)):*
> ```XML
>           <SelectContent
>             // Need to get over the floating web view z-index 200
>             style={{ zIndex: 250 }}
> ```

Better to update the `tw-z-50` in `lib\platform-bible-react\src\components\shadcn-ui\select.tsx` to `tw-z-\[250\]`. That will fix this issue for all usages of `SelectContent`
___
*[`extensions/src/paratext-registration/src/internet-settings.web-view.tsx` line 295 at r2](https://reviewable.io/reviews/paranext/paranext-core/1837#-OZnvVq45daJs_5Qy6L8:-OZnvVq45daJs_5Qy6L9:b8lf8dw) ([raw file](https://github.com/paranext/paranext-core/blob/b82017687b05b21045d1aee562a270708fe5ae6a/extensions/src/paratext-registration/src/internet-settings.web-view.tsx#L295)):*
> ```XML
>           <Alert ref={scrollToRef} variant="destructive">
>             <AlertCircle className="tw-h-4 tw-w-4" />
>             <AlertTitle>{localizedStrings['%general_error_title%']}</AlertTitle>
> ```

I don't think this key is available here


<!-- Sent from Reviewable.io -->

captaincrazybro commented:
*[`extensions/src/paratext-registration/contributions/localizedStrings.json` line 57 at r2](https://reviewable.io/reviews/paranext/paranext-core/1837#-OZnc8DlEPFqsEoUYrzU:-OZo1hca474owasXxBxe:bmr0asn) ([raw file](https://github.com/paranext/paranext-core/blob/b82017687b05b21045d1aee562a270708fe5ae6a/extensions/src/paratext-registration/contributions/localizedStrings.json#L57)):*
<details><summary><i>Previously, rolfheij-sil (Rolf Heij) wrote…</i></summary><blockquote>

UX wants us to only use capitalization on the first word. So I believe this should be `Internet settings`

I notice that many strings have words with capitals mid-sentence. Maybe best to ask UX to review all of these strings. Or were they provided by UX?
</blockquote></details>

Yes, most of this stuff I copied from the Figma designs they made: https://www.figma.com/design/IlfaNPYLjHFFMnR8V7oD3j/Registration-PT-3078?node-id=2-162&p=f&t=9qLPFZTlIPu0IPda-0


<!-- Sent from Reviewable.io -->

captaincrazybro commented:
*[`extensions/src/paratext-registration/contributions/localizedStrings.json` line 115 at r2](https://reviewable.io/reviews/paranext/paranext-core/1837#-OZncy-DAeFKe-Pgg9MZ:-OZo2g4d5isw5TvIpzTy:bh8xzid) ([raw file](https://github.com/paranext/paranext-core/blob/b82017687b05b21045d1aee562a270708fe5ae6a/extensions/src/paratext-registration/contributions/localizedStrings.json#L115)):*
<details><summary><i>Previously, rolfheij-sil (Rolf Heij) wrote…</i></summary><blockquote>

Are you sure that this is correct Spanish? Or was this added by AI?
</blockquote></details>

I have no idea :)
I just put something in there from Google Translate. Maybe that's not the best idea? Who goes in afterward and adds in the Spanish?


<!-- Sent from Reviewable.io -->

captaincrazybro commented:
*[`extensions/src/paratext-registration/src/types/paratext-registration.d.ts` line 79 at r1](https://reviewable.io/reviews/paranext/paranext-core/1837#-OZikYG42sthE6U1kcXq:-OZo3wqbDH_ywcBr9SCc:b8qesem) ([raw file](https://github.com/paranext/paranext-core/blob/4733ce2db1fe2bf65f703afd480f742f4acf09b3/extensions/src/paratext-registration/src/types/paratext-registration.d.ts#L79)):*
<details><summary><i>Previously, rolfheij-sil (Rolf Heij) wrote…</i></summary><blockquote>

Agreed, make sense to put them in the same webview
</blockquote></details>

UX specifically wants two separate popups that separate text (but the same registration form) and different title and tooltip. So I figured the easiest way to do that (especially with the different titles and tooltips) would be to make two separate web views.


<!-- Sent from Reviewable.io -->

captaincrazybro commented:
*[`extensions/src/paratext-registration/src/internet-settings.web-view.tsx` line 163 at r2](https://reviewable.io/reviews/paranext/paranext-core/1837#-OZnypgm9p4EoL61p9en:-OZo96PW7t7TtNaOjx20:b-dzi2ti) ([raw file](https://github.com/paranext/paranext-core/blob/b82017687b05b21045d1aee562a270708fe5ae6a/extensions/src/paratext-registration/src/internet-settings.web-view.tsx#L163)):*
<details><summary><i>Previously, rolfheij-sil (Rolf Heij) wrote…</i></summary><blockquote>

This useEffect needs to clean up the `saveTimout` timeout when the component unmounts while the timer is still running to prevent leaking memory
</blockquote></details>

How would I do that?


<!-- Sent from Reviewable.io -->

captaincrazybro commented:
*[`extensions/src/paratext-registration/src/internet-settings.web-view.tsx` line 242 at r2](https://reviewable.io/reviews/paranext/paranext-core/1837#-OZnwBlx02dEy65HJF6i:-OZo9Cep8xMnbgkjm3md:bnenkav) ([raw file](https://github.com/paranext/paranext-core/blob/b82017687b05b21045d1aee562a270708fe5ae6a/extensions/src/paratext-registration/src/internet-settings.web-view.tsx#L242)):*
<details><summary><i>Previously, rolfheij-sil (Rolf Heij) wrote…</i></summary><blockquote>

parseInt can fail and return NaN. That edge case should be handled too
</blockquote></details>

I didn't actually make this change (I just reused the original code). But I can try to add something


<!-- Sent from Reviewable.io -->

captaincrazybro commented:
*[`extensions/src/paratext-registration/src/utils/helpers.ts` line 1 at r2](https://reviewable.io/reviews/paranext/paranext-core/1837#-OZnkSXqBOF1JDi-PESK:-OZoDKON2_VIa7ayHdbD:bwtvw2o) ([raw file](https://github.com/paranext/paranext-core/blob/b82017687b05b21045d1aee562a270708fe5ae6a/extensions/src/paratext-registration/src/utils/helpers.ts#L1)):*
<details><summary><i>Previously, rolfheij-sil (Rolf Heij) wrote…</i></summary><blockquote>

What do we need this for? I see it is used on `<Alert>` but I don't understand why we need it. Can we try removing it?
</blockquote></details>

This was here in the original code.


<!-- Sent from Reviewable.io -->

captaincrazybro commented:
*[`extensions/src/paratext-registration/src/components/section.component.tsx` line 1 at r2](https://reviewable.io/reviews/paranext/paranext-core/1837#-OZnl4k4C92nBTbSTbGk:-OZoFVkuCIX9245CPcVh:bgaga0z) ([raw file](https://github.com/paranext/paranext-core/blob/b82017687b05b21045d1aee562a270708fe5ae6a/extensions/src/paratext-registration/src/components/section.component.tsx#L1)):*
<details><summary><i>Previously, rolfheij-sil (Rolf Heij) wrote…</i></summary><blockquote>

This component does not make sense to me. Essentially it's just a `div` with `tw-m-2`. Why not just add that Tailwind class instead of creating an entire component for it?
</blockquote></details>

This was also just in the original code. Should I still try to remove this? The other thing is that this `Section` component is used in a lot of places.


<!-- Sent from Reviewable.io -->

captaincrazybro commented:
*[`extensions/src/paratext-registration/src/components/grid.component.tsx` line 1 at r2](https://reviewable.io/reviews/paranext/paranext-core/1837#-OZnlMJfEgokUJ4FnTxZ:-OZoHcf4683tieaeqfJE:b-gw2qhs) ([raw file](https://github.com/paranext/paranext-core/blob/b82017687b05b21045d1aee562a270708fe5ae6a/extensions/src/paratext-registration/src/components/grid.component.tsx#L1)):*
<details><summary><i>Previously, captaincrazybro (Cqptain) wrote…</i></summary><blockquote>

Again, this was just a component that already existed in the code and I exported it so that both the internet settings and paratext registration web views could access it.
</blockquote></details>

But it looks like this is specific and specialized enough that it probably should just be here in the extension.


<!-- Sent from Reviewable.io -->

captaincrazybro commented:
*[`extensions/src/paratext-registration/src/components/registration-form.component.tsx` line 25 at r2](https://reviewable.io/reviews/paranext/paranext-core/1837#-OZnm4pzEYeLE2G8ImIX:-OZoDoCVDqt1NJ6iTJUf:b-896fix) ([raw file](https://github.com/paranext/paranext-core/blob/b82017687b05b21045d1aee562a270708fe5ae6a/extensions/src/paratext-registration/src/components/registration-form.component.tsx#L25)):*
<details><summary><i>Previously, rolfheij-sil (Rolf Heij) wrote…</i></summary><blockquote>

This is currently not being used
</blockquote></details>

Done.


<!-- Sent from Reviewable.io -->

captaincrazybro reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/paranext/paranext-core/1837)* status: 13 of 19 files reviewed, 1 unresolved discussion (waiting on @rolfheij-sil and @tjcouch-sil)

___
*[`extensions/src/paratext-registration/contributions/localizedStrings.json` line 83 at r1](https://reviewable.io/reviews/paranext/paranext-core/1837#-OZilw0uBAVsZlSh4gey:-O_Ch8Zw4ZXdoqyi_mke:b-896fix) ([raw file](https://github.com/paranext/paranext-core/blob/373afef728852e12e24ac8f6d15b813099442073/extensions/src/paratext-registration/contributions/localizedStrings.json#L83)):*
<details><summary><i>Previously, tjcouch-sil (TJ Couch) wrote…</i></summary><blockquote>

Right, but the content of the string changed from "Paratext Registration Information" to "Paratext Registration". It needs to be deprecated and a new string key made.
</blockquote></details>

Done.
___
*[`extensions/src/paratext-registration/src/utils/helpers.ts` line 1 at r2](https://reviewable.io/reviews/paranext/paranext-core/1837#-OZnkSXqBOF1JDi-PESK:-O_CfK7M46KIgCos5unZ:b-896fix) ([raw file](https://github.com/paranext/paranext-core/blob/b82017687b05b21045d1aee562a270708fe5ae6a/extensions/src/paratext-registration/src/utils/helpers.ts#L1)):*
<details><summary><i>Previously, tjcouch-sil (TJ Couch) wrote…</i></summary><blockquote>

When the registration WebView is short, the alert is off the screen, so it's confusing for the user when their stuff doesn't work. So I had it scroll to the error message when they run into an error so they can see what's wrong. I think that's why this was here. Without any context, I'd probably recommend keeping it in since we still don't have a minimum height on the WebView. Probably not a huge deal either way.
</blockquote></details>

Done.


<!-- Sent from Reviewable.io -->

sarahxy commented:
Updated the error validation copy here: https://www.figma.com/design/IlfaNPYLjHFFMnR8V7oD3j/Registration-PT-3078?node-id=914-1570&t=Q4qDSmCNUKZMPCNr-1

Some more nitpicky notes on styling:
- No field should be red on initial start / by default
- Error validation should not show up while the user is typing / entering information in the field, but only when the user clicks away or is idle (since our primary action button is not activated until the fields are valid
- Title case for "Registration Name" and "Registration Code" fields (this also matches the email). So not "Registration name" but "Registration Name"
- Can we put the form validation messages directly under their fields, and put more margin / padding between the error validation messages and the primary action buttons? (Right now they seem equidistant from the field and the primary action buttons.)

rolfheij-sil reviewed (dismissed):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>
Thanks for the hard work and dedication!

> @rolfheij-sil reviewed 6 of 6 files at r9, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/paranext/paranext-core/1837#-:-O_pi_3WERJmIZ4AUZCb:b-1uwjum)* status: <span>:shipit:</span> complete! all files reviewed, all discussions resolved



<!-- Sent from Reviewable.io -->

captaincrazybro reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/paranext/paranext-core/1837)* status: 11 of 20 files reviewed, 4 unresolved discussions (waiting on @rolfheij-sil and @tjcouch-sil)

___
*[`assets/localization/metadata.json` line 75 at r11](https://reviewable.io/reviews/paranext/paranext-core/1837#-O_wlbPm9NjWdf8NMVq4:-O_wqNmfDRdookK7KlAs:b-896fix) ([raw file](https://github.com/paranext/paranext-core/blob/5815a1b5a94eeecd04e4752713af7e763ec2c1cd/assets/localization/metadata.json#L75)):*
<details><summary><i>Previously, rolfheij-sil (Rolf Heij) wrote…</i></summary><blockquote>

The message says the new 'locale' is `%data_loading_error_paratextData_internet_disabled%`, but that's not true. That's the key we're deprecating.

Also I don't think 'locale' is the right word here. You probably mean '(localization) key'
</blockquote></details>

Done.
___
*[`extensions/src/paratext-registration/contributions/localizedStrings.json` line 78 at r9](https://reviewable.io/reviews/paranext/paranext-core/1837#-O_vvkYw466aE8uvJMn5:-O_wqQlr7nkAbkbvfAcC:b-896fix) ([raw file](https://github.com/paranext/paranext-core/blob/17b6ab121cdd78ce3a36e411f15d95bc85e2bdf0/extensions/src/paratext-registration/contributions/localizedStrings.json#L78)):*
<details><summary><i>Previously, tjcouch-sil (TJ Couch) wrote…</i></summary><blockquote>

I'm guessing this change from "Updated registration information" to "Registration information updated" was specified by @paranext/paranext-ux . Could someone from UX please inform us regarding whether you consider this change to be effectively changing the meaning of the string? I'm guessing not, but I want to make sure. Here's the snippet from the [Code Style Guide](https://github.com/paranext/paranext/wiki/Code-Style-Guide#localized-strings-guidelines) that is relevant:

> **Don't change a localized string value's meaning**: You can correct grammar or punctuation mistakes, but don't change the meaning of a string. You may break other uses of the string if you do. Rather, create a new string if you need to change a string's meaning. If you change all known uses of that key in the code that provides that key, consider [deprecating](https://github.com/paranext/paranext-core/blob/197c0893752324fbb587c50e2f98699969cf39cc/assets/localization/metadata.json#L13) the key to indicate it is old and not to be used anymore.

Another way of looking at this would be to ask whether this English change reasonably ought to be reflected in other languages or if it's just an English-specific grammar thing.
</blockquote></details>

Done.


<!-- Sent from Reviewable.io -->

rolfheij-sil reviewed (approved):
> @rolfheij-sil reviewed 9 of 9 files at r12, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/paranext/paranext-core/1837)* status: all files reviewed, 1 unresolved discussion (waiting on @tjcouch-sil)

___
*[`assets/localization/en.json` line 10 at r11](https://reviewable.io/reviews/paranext/paranext-core/1837#-O_wlSUHBMMw57q1hmYn:-O_xSRhS5K6cPoNcI14e:b-x8i3jj) ([raw file](https://github.com/paranext/paranext-core/blob/5815a1b5a94eeecd04e4752713af7e763ec2c1cd/assets/localization/en.json#L10)):*
<details><summary><i>Previously, tjcouch-sil (TJ Couch) wrote…</i></summary><blockquote>

Honestly if the key name still fits and the old string should be deprecated in favor of the new string in all circumstances, I think it should just say `_2`. This way, the relationship between them is very clear in that the updated one is supposed to be a complete replacement based on updated functionality. If something changed in the string that would affect the key name, I'd agree that a key name would make sense.

However, I could kinda see the argument if the first string was still useful for something and ought to stick around. But in this case, the old string is deprecated because it has a very particular meaning that is tied to an implementation that no longer exists.

What do you think?
</blockquote></details>

Hm yeah I guess that makes sense.. `_2` is fine then
___
*[`assets/localization/metadata.json` line 75 at r11](https://reviewable.io/reviews/paranext/paranext-core/1837#-O_wlbPm9NjWdf8NMVq4:-O_xSqAU9PbbegPizIA4:bkjsd4g) ([raw file](https://github.com/paranext/paranext-core/blob/5815a1b5a94eeecd04e4752713af7e763ec2c1cd/assets/localization/metadata.json#L75)):*
<details><summary><i>Previously, captaincrazybro (Cqptain) wrote…</i></summary><blockquote>

Done.
</blockquote></details>

Thanks for fixing. No longer blocking.

Just for my understanding: 'locale' basically means 'set of parameters that define the user's language, country, and cultural conventions for how data should be formatted and displayed', right? So refering to the key of a localizable string as a 'locale' doesn't make sense, right?


<!-- Sent from Reviewable.io -->

sarahxy commented:
> > @tjcouch-sil reviewed 1 of 12 files at r5, 1 of 6 files at r9.
> > _[Reviewable](https://reviewable.io/reviews/paranext/paranext-core/1837)_ status: all files reviewed, 2 unresolved discussions
> 
> _[`extensions/src/paratext-registration/contributions/localizedStrings.json` line 78 at r9](https://reviewable.io/reviews/paranext/paranext-core/1837#-O_vvkYw466aE8uvJMn5:-O_vvkYx6DLGQT5XP5AC:bqnam31) ([raw file](https://github.com/paranext/paranext-core/blob/17b6ab121cdd78ce3a36e411f15d95bc85e2bdf0/extensions/src/paratext-registration/contributions/localizedStrings.json#L78)):_ I'm guessing this change from "Updated registration information" to "Registration information updated" was specified by @paranext/paranext-ux . Could someone from UX please inform us regarding whether you consider this change to be effectively changing the meaning of the string? I'm guessing not, but I want to make sure. Here's the snippet from the [Code Style Guide](https://github.com/paranext/paranext/wiki/Code-Style-Guide#localized-strings-guidelines) that is relevant:
> 
> > **Don't change a localized string value's meaning**: You can correct grammar or punctuation mistakes, but don't change the meaning of a string. You may break other uses of the string if you do. Rather, create a new string if you need to change a string's meaning. If you change all known uses of that key in the code that provides that key, consider [deprecating](https://github.com/paranext/paranext-core/blob/197c0893752324fbb587c50e2f98699969cf39cc/assets/localization/metadata.json#L13) the key to indicate it is old and not to be used anymore.
> 
> Another way of looking at this would be to ask whether this English change reasonably ought to be reflected in other languages or if it's just an English-specific grammar thing.
> 
> _[`assets/localization/en.json` line 9 at r9](https://reviewable.io/reviews/paranext/paranext-core/1837#-O_vuquAAuxPsu6AT_9d:-O_vuquAAuxPsu6AT_9e:b-p1uspg) ([raw file](https://github.com/paranext/paranext-core/blob/17b6ab121cdd78ce3a36e411f15d95bc85e2bdf0/assets/localization/en.json#L9)):_
> 
> > ```json
> >   "%ariaLabel_opensInBrowser%": "Opens externally in a browser window",
> >   "%data_loading_error_paratextData_auth_failure%": "Paratext Registration is not valid. Please update your registration information and try again.",
> >   "%data_loading_error_paratextData_internet_disabled%": "Internet access is disabled in “Internet Settings”. Please enable it and try again.",
> > ```
> 
> This is another existing string that needs to be deprecated and replaced.

@tjcouch-sil hi~ just FYI i was the one who changed it. it looks like it was resolved here and also on Discord, but for context i made the change bc most of the other error messages have that format ("Registration information ___"), so i tried to make it consistent. that was all!

tjcouch-sil reviewed (commented):
> @tjcouch-sil reviewed 2 of 9 files at r12.
> *[Reviewable](https://reviewable.io/reviews/paranext/paranext-core/1837)* status: all files reviewed, 1 unresolved discussion (waiting on @captaincrazybro)

___
*[`assets/localization/en.json` line 9 at r9](https://reviewable.io/reviews/paranext/paranext-core/1837#-O_vuquAAuxPsu6AT_9d:-Oa0Me7b2i4Hjd5hNMW5:b-txbfow) ([raw file](https://github.com/paranext/paranext-core/blob/17b6ab121cdd78ce3a36e411f15d95bc85e2bdf0/assets/localization/en.json#L9)):*
<details><summary><i>Previously, sarahxy (Sarah) wrote…</i></summary><blockquote>

@tjcouch-sil hi~ just FYI i was the one who changed it. it looks like it was resolved here and also on Discord, but for context i made the change bc most of the other error messages have that format ("Registration information ___"), so i tried to make it consistent. that was all!
</blockquote></details>

Sounds great. Thanks!
___
*[`assets/localization/metadata.json` line 75 at r11](https://reviewable.io/reviews/paranext/paranext-core/1837#-O_wlbPm9NjWdf8NMVq4:-Oa0Mt0-CSYYmdMTTmaX:b-i24ecm) ([raw file](https://github.com/paranext/paranext-core/blob/5815a1b5a94eeecd04e4752713af7e763ec2c1cd/assets/localization/metadata.json#L75)):*
<details><summary><i>Previously, rolfheij-sil (Rolf Heij) wrote…</i></summary><blockquote>

Thanks for fixing. No longer blocking.

Just for my understanding: 'locale' basically means 'set of parameters that define the user's language, country, and cultural conventions for how data should be formatted and displayed', right? So refering to the key of a localizable string as a 'locale' doesn't make sense, right?
</blockquote></details>

Yes, I would expect this to say "so the localized string needs to indicate that" and "the new localized string key is ___"


<!-- Sent from Reviewable.io -->

rolfheij-sil reviewed (approved):
> @rolfheij-sil reviewed 6 of 6 files at r13, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/paranext/paranext-core/1837)* status: <span>:shipit:</span> complete! all files reviewed, all discussions resolved

___
*[`assets/localization/metadata.json` line 75 at r11](https://reviewable.io/reviews/paranext/paranext-core/1837#-O_wlbPm9NjWdf8NMVq4:-Oa0Ret48D4bhU28XZbp:b-8takfq) ([raw file](https://github.com/paranext/paranext-core/blob/5815a1b5a94eeecd04e4752713af7e763ec2c1cd/assets/localization/metadata.json#L75)):*
<details><summary><i>Previously, tjcouch-sil (TJ Couch) wrote…</i></summary><blockquote>

Yes, I would expect this to say "so the localized string needs to indicate that" and "the new localized string key is ___"
</blockquote></details>

You forgot the word 'key' :sweat_smile: 


<!-- Sent from Reviewable.io -->
